src/main/java/ru/timeconqueror/tcneiadditions/util/TCUtil.java

```diff
@@ -127,4 +134,100 @@ public static List<ItemStack> getOreDictionaryMatchingItemsForInfusion(ItemStack
         }
         return Collections.singletonList(stack);
     }
+
+    public static List<String> getResearchPrerequisites(List<String> list, ResearchItem researchItem) {
+        if (researchItem == null) return list;
+        String playerName = Minecraft.getMinecraft().getSession().getUsername();
+        // Parent research
+        if (researchItem.parents != null && researchItem.parents.length != 0) {
+            list.add(StatCollector.translateToLocal("tcneiadditions.research.prerequisites.parents") + ":");
+            int needResearch = 0;
+            for (String researchKey : researchItem.parents) {
+                String researchName =
+                        ResearchCategories.getCategoryName(ResearchCategories.getResearch(researchKey).category) + " : "
+                                + ResearchCategories.getResearch(researchKey).getName();
+                if (ResearchManager.isResearchComplete(playerName, researchKey)) {
+                    if (researchItem.parents.length <= 10) {
+                        researchName = EnumChatFormatting.GREEN + "" + EnumChatFormatting.STRIKETHROUGH + researchName;
+                        list.add(EnumChatFormatting.RESET + "    " + researchName);
+                    }
+                } else {
+                    needResearch++;
+                    researchName = EnumChatFormatting.RED + researchName;
+                    list.add(EnumChatFormatting.RESET + "    " + researchName);
+                }
+            }
+            if (researchItem.parents.length > 10 && needResearch == 0) {
+                list.add(EnumChatFormatting.GREEN
+                        + StatCollector.translateToLocal("tcneiadditions.research.prerequisites.allresearched"));
+            }
+        }
+        // Parent hidden research
+        if (researchItem.parentsHidden != null && researchItem.parentsHidden.length != 0) {
+            int needResearch = 0;
+            list.add(StatCollector.translateToLocal("tcneiadditions.research.prerequisites.parentsHidden") + ":");
+            for (String researchKey : researchItem.parentsHidden) {
```

Comment:
Parent and hidden parent code have common parts, why not extract to another method?